### PR TITLE
FindingsMatcher: Fix broken merging of matched root license findings

### PR DESCRIPTION
**Pull request by fviernau**
The `matchedRootLicenseFindings` were added to `matchedFindings` by
the `+=` operator which does an add-or-replace operation. Effectivly
this discards all copyright statements from the root license file in
case there are unmatched copyrights. Fix this by merging the copyright
statements properly.

While at it use merge() also for merging `matchedFileFindings` for
consistency, even if this is not necessary for fixing this bug.

Signed-off-by: Frank Viernau <frank.viernau@here.com>

### CODE REVIEW

**Comment by sschuberth** [comment]
Commit message: s/Effectivly/Effectively/

File: model/src/main/kotlin/utils/FindingsMatcher.kt [hunk 1]
```diff
@@ -225,3 +225,11 @@ data class FindingsMatcherResult(
 )
 
 private fun TextLocation.directory(): String = path.substringBeforeLast(delimiter = "/", missingDelimiterValue = "")
+
+private fun MutableMap<LicenseFinding, MutableSet<CopyrightFinding>>.merge(
```
Comment (sschuberth) [inline model/src/main/kotlin/utils/FindingsMatcher.kt:229]: How about using the existing `zipWithDefault` extension function for maps instead?

**Comment by fviernau** [comment]
> Commit message: s/Effectivly/Effectively/

done.

File: model/src/main/kotlin/utils/FindingsMatcher.kt (hunk 1 above)
Comment (fviernau) [reply model/src/main/kotlin/utils/FindingsMatcher.kt:229]: That'd require turning `val matchedFindings` into `var matchedFindings`.
I'd guess that the current implementation could be more efficient and I find it readable.
Given that I"d prefer to keep it `as-is`. Makes sense to you?

**Review by mnonnenmacher — APPROVED** [review]

**Pull request merged**